lib/active_job/queue_adapters/solid_queue_ext.rb

```diff
@@ -82,6 +82,12 @@ def discard_job(job, jobs_relation)
     find_solid_queue_job!(job.job_id, jobs_relation).discard
   end
 
+  def dispatch_job(job, jobs_relation)
+    job = find_solid_queue_job!(job.job_id, jobs_relation)
+
+    dispatch_blocked_job_immediately(job) if job.blocked?
```

Comment:
Do you need the `job.blocked?` check here if the `jobs_relation` that you're receiving is already for blocked jobs? That's it, why couldn't this be the same as this method, where we don't check `job.failed?` before calling `retry`: 
```ruby
  def retry_job(job, jobs_relation)
    find_solid_queue_job!(job.job_id, jobs_relation).retry
  end
```